### PR TITLE
remove cloud provider dependencies to pkg/volume

### PR DESCRIPTION
**Pull request by andrewsykim**
Co-authored-by: Weibin Lin <linweibin1@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removes cloud provider dependencies to `pkg/volume/*`. 

Moves all the volume zone label methods from `pkg/volume/util` to `k8s.io/cloud-provider/volume`. Also moves some of the rounding methods there as well, wasn't sure if `k8s.io/cloud-provider/volume` is the right place for those but wasn't sure where else it could go in the meantime. 

**Which issue(s) this PR fixes**:
Part of https://github.com/kubernetes/kubernetes/issues/69585 which is needed to move cloud providers out-of-tree. 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by andrewsykim** [comment]
Blocked on https://github.com/kubernetes/kubernetes/pull/73616

File: staging/src/k8s.io/cloud-provider/volume/constants.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	// ProvisionedVolumeName is the name of a volume in an external cloud
+	// that is being provisioned and thus should be ignored by rest of Kubernetes.
+	ProvisionedVolumeName = "placeholder-for-provisioning"
+
+	// LabelMultiZoneDelimiter is used as a delimiter when multiple zones as specified as part of a PV's label
+	LabelMultiZoneDelimiter = "__"
+
+	// LabelZoneFailureDomain should be removed!
+	// TODO: remove and use from v1 once https://github.com/kubernetes/kubernetes/pull/73616 is merged
+	LabelZoneFailureDomain = "failure-domain.beta.kubernetes.io/zone"
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/volume/constants.go]: This is temporary to skip import restriction check, will remove once https://github.com/kubernetes/kubernetes/pull/73616 is merged.

File: staging/src/k8s.io/cloud-provider/volume/helpers/rounding_test.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
+/*
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/volume/helpers/rounding_test.go]: This is a new file

File: staging/src/k8s.io/cloud-provider/volume/helpers/rounding.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
+/*
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/volume/helpers/rounding.go]: Not sure if volume rounding belongs in this package, open to other suggestions here.

**Comment by msau42** [comment]
/assign @ddebroy 
to help review the volume zone and migration changes

File: staging/src/k8s.io/cloud-provider/volume/errors/errors.go [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2017 The Kubernetes Authors.
+Copyright 2019 The Kubernetes Authors.
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/volume/errors/errors.go]: we should not be updating copyrights.

**Review by thockin — COMMENTED** [review]
One significant point (copyrights).

Overall, fine, and I will approve once above is done, but the reality is that most of this code should be localized within a provider.  It feels like it is trying too hard to be general-purpose. :(

**Comment by andrewsykim** [comment]
Thanks @thockin, comment addressed! 

> but the reality is that most of this code should be localized within a provider. It feels like it is trying too hard to be general-purpose. :(

Agreed, I'll start putting some more thought into how to approach this going forward, the zone labelling logic is used in a bunch of places and it would be good to clean this up.

**Review by feiskyer — COMMENTED** [review]
lgtm (for cloud provider parts)

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**